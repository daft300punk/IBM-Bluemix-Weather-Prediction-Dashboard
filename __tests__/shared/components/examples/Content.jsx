import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import Content from 'components/examples/Content';

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();
  renderer.render(<Content />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
