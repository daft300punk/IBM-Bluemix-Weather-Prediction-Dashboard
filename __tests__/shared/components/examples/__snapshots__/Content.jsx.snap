// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-examples-Content-___style__Content___3X7yY"
>
  <h1>
    Isomorphic ReactJS App
  </h1>
  <p>
    Technological stack includes:
  </p>
  <ul>
    <li>
      Autoprefixer;
    </li>
    <li>
      Babel with latest JS standard support both client- and server-side;
    </li>
    <li>
      ESlint (AirBnB style, run with 
      <code>
        $ npm run lint:js
      </code>
      );
    </li>
    <li>
      ExpressJS server;
    </li>
    <li>
      Font loading (Roboto fonts are included into the repo);
    </li>
    <li>
      Hot reload of JS code and SCSS styles in dev environment (start it with 
      <code>
        $ npm run dev
      </code>
      );
    </li>
    <li>
      Loading of .svg assets as ReactJS components with 
      <code>
        babel-plugin-inline-react-svg
      </code>
      ;
    </li>
    <li>
      ReactJS;
    </li>
    <li>
      React CSS Modules (with 
      <code>
        babel-plugin-react-css-modules
      </code>
      );
    </li>
    <li>
      Redux with Flex Standard Actions, redux-promise middleware, and a custom pattern of server-side data fetching;
    </li>
    <li>
      SCSS styles;
    </li>
    <li>
      Stylefmt;
    </li>
    <li>
      Stylelint for SCSS (standard Stylelint style, run with 
      <code>
        $ npm run lint:scss
      </code>
      ;
    </li>
    <li>
      Webpack;
    </li>
  </ul>
  <h3>
    Misc Examples
  </h3>
  <ul>
    <li>
      <Link
        replace={false}
        to="/examples/css-modules"
      >
        CSS Modules
      </Link>
       - Demo/test of CSS modules in action;
    </li>
    <li>
      <Link
        replace={false}
        to="/examples/data-fetch"
      >
        Data Fetch
      </Link>
       - Demonstrates how data fetching should be implemented in isomorphic way, using Redux with Flux Standard Actions and promise;
    </li>
    <li>
      <Link
        replace={false}
        to="/examples/fonts-test"
      >
        Fonts Test
      </Link>
       - A simple showcase of the fonts included into this repo, and the test of their proper inclusion into the bundle;
    </li>
    <li>
      <Link
        replace={false}
        to="/examples/svg-loading"
      >
        SVG Loading
      </Link>
       - Shows how to load 
      <code>
        .svg
      </code>
       assets with use of 
      <code>
        babel-plugin-inline-react-svg
      </code>
      .
    </li>
    <li>
      <Link
        replace={false}
        to="/examples/themr"
      >
        Themr
      </Link>
       - Test/demo of react-css-themr.
    </li>
  </ul>
</div>
`;
